### ISSUE TITLE
[BUG] not working after upgrading yarn

### ISSUE THREAD

**Issue by eifr** [npm/cli#4184]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

npm cli not available after trying to `npm update --global yarn`.
ran `npm update --global yarn` and after done npm command didn't work as if it is not installed at all. worked only after reinstalling.

### Expected Behavior

npm cli to keep working even after upgrading yarn

### Steps To Reproduce

1. open terminal
2. run `npm update --global yarn`
3. wait until done
4. try to run any `npm` command

### Environment

- npm: 8.3.0
- Node: 16.13.1
- OS: fedora 35
- platform: x1 9th gen
- npm config:
```ini
; "user" config from /home/eifr/.npmrc

//registry.npmjs.org/:_authToken = (protected) 

; node bin location = /home/eifr/.fnm/node-versions/v16.13.1/installation/bin/node
; cwd = /home/eifr
; HOME = /home/eifr
; Run `npm config ls -l` to show all defaults.
```

**Comment by milaninfy** [comment]
**Closing** This appears to be fixed on latest. Feel free to reopen or refile new issue if this is still an issue. 
```sh
~/workarea/rep/keyword-test $ npm update --global yarn

changed 1 package in 1s

30 packages are looking for funding
  run `npm fund` for details
~/workarea/rep/keyword-test $ npm view yarn@latest version
1.22.22
~/workarea/rep/keyword-test $ npm ls --global  
/Users/milaninfy/.nvm/versions/node/v22.14.0/lib
├── corepack@0.31.0
├── lodash@4.17.21
├── node-gyp@11.1.0
├── npm@11.4.1
└── sass@1.87.0
```
